cli/cmdmanager/manager.go

```diff
@@ -0,0 +1,325 @@
+package cmdmanager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/sensu/sensu-go/asset"
+	"github.com/sensu/sensu-go/bonsai"
+	"github.com/sensu/sensu-go/command"
+	"github.com/sensu/sensu-go/system"
+	"github.com/sensu/sensu-go/util/environment"
+	"github.com/sensu/sensu-go/util/path"
+
+	goversion "github.com/hashicorp/go-version"
+	corev2 "github.com/sensu/sensu-go/api/core/v2"
+	bolt "go.etcd.io/bbolt"
+)
+
+const (
+	dbName = "commands.db"
+)
+
+var (
+	commandBucketName = []byte("commands")
+)
+
+type CommandManager struct {
+	assetManager *asset.Manager
+	assetGetter  asset.Getter
+	db           *bolt.DB
+}
+
+type CommandPlugin struct {
+	Alias string       `json:"alias"`
+	Asset corev2.Asset `json:"asset"`
+}
+
+func (p *CommandPlugin) GetObjectMeta() corev2.ObjectMeta {
+	return corev2.ObjectMeta{Name: p.Alias}
+}
+
+func (p *CommandPlugin) RBACName() string {
+	return ""
+}
+
+func (p *CommandPlugin) StorePrefix() string {
+	return ""
+}
+
+func (p *CommandPlugin) URIPath() string {
+	return ""
+}
+
+func (p *CommandPlugin) Validate() error {
+	return nil
+}
+
+func (p *CommandPlugin) SetNamespace(namespace string) {
+	// no-op
+}
+
+func (p *CommandPlugin) SetObjectMeta(meta corev2.ObjectMeta) {
+	// no-op
+}
```

Comment:
What's the purpose of implementing `types.Resource` here?

Reply:
It's necessary to use the table output functions. We should find a way to fix this.